File: hackertracker/src/main/java/com/shortstack/hackertracker/Analytics/FilterEvent.java
package com.shortstack.hackertracker.Analytics;

import com.crashlytics.android.answers.CustomEvent;
import com.shortstack.hackertracker.Model.Filter;

public class FilterEvent extends CustomEvent {

    public FilterEvent( Filter filter ) {
        super(AnalyticsController.Analytics.SCHEDULE_FILTERS.toString());
        String[] array = filter.getTypesArray();

        for (int i = 0; i < array.length; i++) {
            putCustomAttribute(array[i], String.valueOf(true));
        }
    }
}

File: hackertracker/src/main/java/com/shortstack/hackertracker/Analytics/FilterEvent.kt
package com.shortstack.hackertracker.analytics

import com.crashlytics.android.answers.CustomEvent
import com.shortstack.hackertracker.models.Filter

class FilterEvent(filter : Filter) : CustomEvent(AnalyticsController.Analytics.SCHEDULE_FILTERS.toString()) {

    init {
        val array = filter.typesArray

        for (i in array.indices) {
            putCustomAttribute(array[i], true.toString())
        }
    }
}
